From a5b428ee41da428498597d4279ffea4b8cc3495c Mon Sep 17 00:00:00 2001
From: leej3 <leej3@quansight.com>
Date: Wed, 4 May 2022 09:38:55 -0500
Subject: [PATCH] add fix for jpeg boolean type

Since release 1.18.0 some fixes have been made to accommodate more
stringent type checking of boolean values passed to the jpeg API
(introduced in version
8).

This patch is a minimal equivalent fix.  TRUE and FALSE can be
interpreted as either macro definitions or values of the enum boolean
type both defined in the jpeg header files. Depending on a compile time
definition the jconfig.h header may be included as part of these header
files. Macro definitions in downstreams can vary based on these two
situations.  This patch for this library is to define TRUE and FALSE as
macros and then pass a macro that expands to TRUE or FALSE.  Directly
using 1 or 0 fails with a type casting error during compilation
(inferred to be due to 1 and 0 not being in the boolean enum).
---
 DevIL/include/IL/il.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/DevIL/include/IL/il.h b/DevIL/include/IL/il.h
index 47dfcd7..054f437 100644
--- a/DevIL/include/IL/il.h
+++ b/DevIL/include/IL/il.h
@@ -115,8 +115,11 @@ typedef long long unsigned int ILuint64;
 	#define ILconst_string char const *
 #endif //_UNICODE
 
-#define IL_FALSE			0
-#define IL_TRUE				1
+#define FALSE			0
+#define TRUE			1
+#define IL_FALSE			FALSE
+#define IL_TRUE				TRUE
+
 
 //  Matches OpenGL's right now.
 //! Data formats \link Formats Formats\endlink
-- 
2.17.0.windows.1

